svc-bgs-api/src/lib/rabbit_subscriber.rb

```diff
@@ -0,0 +1,88 @@
+# Inspired from https://www.cloudbees.com/blog/writing-microservice-in-ruby
+require 'bunny'
+require 'json'
+require 'logger'
+
+class RabbitSubscriber
+
+  def initialize(bunny_args)
+    @connection = Bunny.new(bunny_args)
+    connect(@connection)
+    @exchanges = {}
+    @queues = {}
+  end
+
+  def connect(connection)
+    attempts ||= 1
+    connection.start
+  rescue => error
+    $logger.error error.message
+    if (attempts += 1) < 5
+      $logger.warn "Retrying to connect to RabbitMQ in 10 seconds"
+      sleep 10
+      retry
+    else
+      raise error
```

Comment:
What happens as a result of this `raise`? Will ruby exit?

Reply:
Yes, the process will exit in 40 seconds if all attempts to connect to RabbitMQ fail.